### PR TITLE
Drawer: increase test coverage in Karma test runner

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] https://github.com/Samsung/TAU/issues/1306
[Problem] Drawer increase test covereage
[Solution]
 - Test for core/Drawer has been added

obraz

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

### CODE REVIEW

File: tests/js/core/widget/core/Drawer/test-data/sample.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<div class="ui-page ui-page-active" id="page-1">
+	<div class="ui-contnent">
+		<div class="ui-drawer" data-position="left" id="drawer-1" data-drag-edge="50">
```
Comment (DohyungLim) [inline tests/js/core/widget/core/Drawer/test-data/sample.html:3]: question.
what is data-drag-edge? and why value 50 fixed?

**Review by DohyungLim — COMMENTED** [review]
1 comment.

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+			// case 1
+			widget._translate(10, 20, 1000);
+
+			assert.equal(widget._state, "sliding", "widget _state set to 'sliding'");
+			// this test depends to web engine because: transform, -webkit-transform
+			//assert.ok(!!element.style.transition, "transform style has been set'");
+
+			// case 2 (duration is 0)
+			widget._state = "not settling";
+			widget._translate(10, 20, 0);
```
Comment (lslachciak) [inline tests/js/core/widget/core/Drawer/Drawer.js:497]: some questions
1. no need to check widget state?
2. shouldn't we check _translatedX here?

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+			});
+			helpers.stub(widget, "_onTransitionEnd", function () {
+				assert.ok(true, "_onTransitionEnd has been called");
+			});
+
+			// case 1
+			widget._translate(10, 20, 1000);
+
+			assert.equal(widget._state, "sliding", "widget _state set to 'sliding'");
+			// this test depends to web engine because: transform, -webkit-transform
```
Comment (lslachciak) [inline tests/js/core/widget/core/Drawer/Drawer.js]: if test is not relevant why keep it?

File: tests/js/core/widget/core/Drawer/test-data/sample.html (hunk 1 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/test-data/sample.html:3]: I copied this `html` from examples, but this option is defined in Drawer widget (https://github.com/Samsung/TAU/blob/dc24f0903ccda2c8c0fb34b4f9696ea9c55816c2/src/js/core/widget/core/Drawer.js#L126)

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 3 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: I'll remove it.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 2 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js:497]: 1. I think we don't need check widget state. It has already been checked in previous case.
I can add this check if you wish.
2. We have to call `_translate` with duration `0` to check if the method `_onTransitionEnd` will be call.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 3 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: Removed.

**Review by lslachciak — APPROVED** [review]

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+				element = document.getElementById("drawer-1"),
+				result;
+
+			widget.options.overlay = true;
+
+			result = widget._build(element);
+
+			assert.equal(result, element, "method returns widget based element");
+			assert.ok(element.classList.contains("ui-drawer"), "widget class has been added to element");
+			assert.equal(element.style.top, "0px", "widget class has been added to element");
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js]: Needs to check the description.

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+			});
+
+			result = widget._init(element);
+
+			assert.equal(result, element, "method returns widget based element");
+			assert.equal(widget._ui.drawerPage, document.getElementById("page-1"), "page has been cached in widget");
+			assert.equal(widget._ui.drawerPage.style.overflowX, "hidden", "overflowX is hidden for page");
+		});
+
+		test("Test of handleEvent drawer", 34, function (assert) {
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js:56]: Is correct count(34) for the 'test'?

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+			widget.handleEvent(testEvent);
+
+			testEvent.type = "msTransitionEnd";
+			widget.handleEvent(testEvent);
+
+			testEvent.type = "resize";
+			widget.handleEvent(testEvent);
+		});
+
+		test("Test of _onTransitionEnd drawer", 12, function (assert) {
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js:159]: ditto. 12.

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 7]
```diff
@@ -0,0 +1,684 @@
...
+
+			// case right
+			widget._state = "not opened";
+			widget.options = {
+				position: "right",
+				width: window.innerWidth
+			};
+			helpers.stub(widget, "_translate", function (x, y, duration) {
+				assert.ok(true, "_translate has been called");
+				assert.equal(x, 0, "x has been provided as window.innerWidth");
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js]: Is the '0' correct value for checking window.innerWidth?

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 8]
```diff
@@ -0,0 +1,684 @@
...
+			// case down
+			widget._state = "not opened";
+			widget.options = {
+				position: "down",
+				height: window.innerHeight
+			};
+			helpers.stub(widget, "_translate", function (x, y, duration) {
+				assert.ok(true, "_translate has been called");
+				assert.equal(x, 0, "x has been provided as 0");
+				assert.equal(y, 0, "y has been provided as window.innerHeight");
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js]: ditto. Needs to check the 'y'.

File: tests/js/core/widget/core/Drawer/Drawer.js [hunk 9]
```diff
@@ -0,0 +1,684 @@
...
+			var widget = new Drawer();
+
+			helpers.stub(widget, "_refresh", function () {
+				assert.ok(true, "_refresh has been called");
+			});
+
+			widget._onResize();
+		});
+
+		test("Test of _onSwipe drawer", 16, function (assert) {
```
Comment (HunseopJeong) [inline tests/js/core/widget/core/Drawer/Drawer.js:369]: ditto. 16

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 5 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js:56]: This is correct.
`widget.handleEvent(testEvent)` has been called 17 times and each request run 2 assertions.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 6 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js:159]: I think this is correct.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 7 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: I'll change this.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 9 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js:369]: I think this is correct.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: Done.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 7 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: Done.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 8 above)
Comment (TomaszLukawskiSam) [reply tests/js/core/widget/core/Drawer/Drawer.js]: Done.

File: tests/js/core/widget/core/Drawer/Drawer.js (hunk 5 above)
Comment (HunseopJeong) [reply tests/js/core/widget/core/Drawer/Drawer.js:56]: Okay!

**Review by HunseopJeong — APPROVED** [review]
LGTM

**Pull request merged**